test/hotspot/jtreg/compiler/lib/ir_framework/test/IREncodingPrinter.java

```diff
@@ -176,6 +203,62 @@ private boolean hasAllRequiredFlags(String[] andRules, String ruleType) {
         return returnValue;
     }
 
+    private boolean hasRequiredFeaturesAnd(String[] andRules, String ruleType) {
+        for (int i = 0; i < andRules.length; i++) {
+            String feature = andRules[i];
+            i++;
+            String value = andRules[i];
+            TestFormat.check((value.contains("true") || value.contains("false")), "Incorrect value in " + ruleType + failAt());
+            if (!checkTargetFeature(feature, value)) {
+                // Rule will not be applied but keep processing the other flags to verify that they are same.
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private boolean hasRequiredFeaturesOr(String[] orRules, String ruleType) {
+        boolean returnValue = false;
+        for (int i = 0; i < orRules.length; i++) {
+            String feature = orRules[i];
+            i++;
+            String value = orRules[i];
+            TestFormat.check((value.contains("true") || value.contains("false")), "Incorrect value in " + ruleType + failAt());
+            returnValue |= checkTargetFeature(feature, value);
+        }
+        return returnValue;
+    }
+
+    private boolean checkTargetFeature(String feature, String value) {
+        String s = WHITE_BOX.getCPUFeatures();
```

Comment:
I suggest to name it `cpuFeatures` instead:
```suggestion
        String cpuFeatures = WHITE_BOX.getCPUFeatures();
```